client/src/AttendanceView/AttendanceView.js

```diff
@@ -136,11 +145,20 @@ export function AttendanceView() {
     setUpdatedAttendanceData(updatedData)
   }
 
+  const [showWeekends, setShowWeekends] = useState(false)
+
+  useEffect(() => {
+    setColumns(generateColumns())
+    // eslint-disable-next-line
+  }, [showWeekends])
+
   // create seven columns for each day of the week
   const generateColumns = () => {
     const dateColumns = []
+    var i = showWeekends ? 0 : 1
```

Comment:
Don't use var. You should be able to use let instead.

Reply:
Done